File: fds-backend/user-management-service/src/main/java/com/yokke/usermanagement/auth/request/LoginRequest.java
package com.yokke.usermanagement.auth.request;

import com.yokke.usermanagement.auth.validation.Lowercase;
import com.yokke.usermanagement.auth.validation.NoWhitespace;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class LoginRequest {

    String password;
    private String email;
    @Size(max = 50)
    @NotNull
    @NotBlank
    @Lowercase
    @NoWhitespace
    private String username;
}

File: fds-dashboard/src/packages/user-management-service/src/auth/request/login.request.ts
import * as z from "zod";

export type LoginRequest = {
  password: string;
  email: string | null;
  username: string;
};
export const LoginRequestSchema = z.object({
  password: z.string(),
  username: z.string(),
});
export type LoginRequestZod = z.infer<typeof LoginRequestSchema>;
